src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -152,7 +152,11 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
 
             if (visualStudioWorkspaceOpt != null)
             {
-                this.EditAndContinueImplOpt = new VsENCRebuildableProjectImpl(this);
+                if (Language == LanguageNames.CSharp || Language == LanguageNames.VisualBasic)
```

Comment:
Why not just put the initialization of EditAndCotntinueImplOpt in the derived types?